docs/src/modules/spring/pages/event-sourced-entities.adoc

```diff
@@ -0,0 +1,160 @@
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/ShoppingCartService.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartService.java[tag=addItem]
+----
+<1> The validation ensures the quantity of items added is greater than zero and it fails for calls with illegal values by returning an `Effect` with `effects().error`.
+<2> From the current incoming `LineItem` we create a new `ItemAdded` event representing the change of the cart.
+<3> We store the event by returning an `Effect` with `effects().emitEvent`.
+<4> The acknowledgment that the command was successfully processed is only sent if the event was successfully stored, otherwise there will be an error reply.
+<5> Event handler needs to be marked with `@EventHandler` and receive a single parameter type matching the event type produced (i.e. `ItemAdded`).
+<6> Return the new state to be stored - the logic for state transition is defined inside the `ShoppingCart` domain model.
+
+
+NOTE: There needs to be one event handler declared per each type of event the ES entity emits (e.g. `itemAdded` receives a parameter of type `ItemAdded`, the same type emitted in `addItem` command handler).
+
+As mentioned above, the business logic that allows us to transition between states was placed on the domain model as seen below:
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=itemAdded]
+----
+<1> For an existing item, we will make sure to sum the existing quantity with the incoming one.
+<2> Returns an update list of items without the existing item.
+<3> Adds the update item to the shopping cart.
+<4> Returns a new instance of the shopping cart with the updated line items.
+
+- the `emptyState()` override is optional. If not using it, be careful to deal with a currentState() with a `null` value when receiving the first command
+
+
+=== Retrieving state
+
+To have access to the current state of the entity we can use `currentState()` as you have probably noticed from the examples above. However, what if this is the first command we are receiving for this entity? The following example shows the implementation of the read-only command handler (accessed through `GET /cart/myCarId`):
+
+[source,java,indent=0]
+.src/main/java/com/example/shoppingcart/ShoppingCartService.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartService.java[tag=getCart]
+----
+<1> Stores the `entityId` on an internal attribute so we can use it later.
+<2> Provides initial state --overriding `emptyState()` is optional but if not doing it, be careful to deal with a currentState() with a `null` value when receiving the first command.
```

Comment:
```suggestion
<2> Provides initial state -- overriding `emptyState()` is optional but if not doing it, be careful to deal with a currentState() with a `null` value when receiving the first command.
```

Reply:
This is actually a Vale rule... it throws an error if using the suggested version (which is also what I was using): ` 116:1  error  Remove the spaces around ' —'.  Lightbend.Dashes`

however, I just checked and it is ok with using "state--overriding" so I will probably update to that since seems to be the format being used elsewhere in the docs..